Ag 13351 Framework selectors

kylerphillips opened the pull request:
https://github.com/user-attachments/assets/7ad8df15-d17d-4c19-871a-b93bc7109bd3


https://github.com/user-attachments/assets/da8281f4-835c-4158-8518-085bdd2c4f3a

taktran committed:
AG-13353 - Allow hero grid to be smaller

taktran committed:
AG-13353 - Fix hero grid dark mode

Need to load only on react so darkmode gets passed through properly

kylerphillips committed:
AG-13351 Add framework selector dropdown to InstallText

taktran committed:
AG-13353 - Add loading spinner to hero grid

kylerphillips committed:
AG-13351 Add frameworks to buttons

taktran committed:
AG-13353 - Add hero links

kylerphillips committed:
AG-13351 Add framework option to landing page section

kylerphillips committed:
AG-13351 Add framework buttons to landing page section

kylerphillips committed:
AG-13351 Add padding to copy code

kylerphillips committed:
AG-13351 Add dropdown in landing page section

kylerphillips committed:
AG-13351 Framework select changes

kylerphillips committed:
AG-1331 Make framework text clickable not dropdown

kylerphillips committed:
AG-13351 Update current framework from storage

kylerphillips committed:
AG-13351 Map frameworks to internal frameworks

kylerphillips committed:
AG-13351 Update framework link

kylerphillips committed:
AG-13351 Update ctaUrl for integrated charts

kylerphillips committed:
AG-13351 Update install text for vue

kylerphillips committed:
AG-13351 Fix monospace

kylerphillips committed:
AG-13351 Fix z-index of automated demo and framework overlay

kylerphillips committed:
AG-13351 Update dropdown active color

kylerphillips committed:
AG-13351 Revert back old buttons

taktran commented on external/ag-website-shared/src/components/landing-pages/LandingPageSection.tsx:
```diff
@@ -29,7 +57,55 @@ export const LandingPageSection: FunctionComponent<Props> = ({
     children,
+    isFramework = false,
 }) => {
+    const framework = useFrameworkFromStore(); // Get the framework from the store
```
We can probably modify `useFrameworkFromStore` to return `internalFramework` along with `framework` here, since `useFrameworkFromStore` already has that information. That means we don't need `mapFrameworkToInternalFramework` (as it's missing the react mapping).

taktran commented on documentation/ag-grid-docs/src/components/framework-landing-pages/InstallText.tsx:
```diff
@@ -2,27 +2,137 @@ import { Icon } from '@ag-website-shared/components/icon/Icon';
 import { useRef, useState } from 'react';
 
 import styles from './InstallText.module.scss';
+import AngularIcon from './angular.svg?react';
```
Why not use `@ag-website-shared/images/inline-svgs/*.svg?react';` here as well?

taktran replied:
Updated

kylerphillips committed:
AG-13351 Fix opacity for chevron dark mode

taktran committed:
AG-13351 - Add container for finance hero

markdurrant committed:
AG-13351 - Tweak hero grid links copy & style

markdurrant committed:
AG-13351 - Tweak hero grid layout

taktran committed:
AG-13351 - Add mobile view for hero grid

Update column defs to be flex for small screen sizes and smaller min width for medium screen sizes

kylerphillips committed:
AG-13351 Small responsive fix quotes

markdurrant committed:
AG-13351 - Tweak CTA styling

markdurrant committed:
AG-13351 - Tweak personalise example grid

markdurrant committed:
AG-13351 - Add more differences to theme change example

markdurrant committed:
AG-13351 - Add project names to showcase section

markdurrant committed:
AG-13351 - Move personalise section between automated examples

markdurrant committed:
AG-13351 - Refactor hero layout

markdurrant committed:
AG-13351 - Improve hero loading section layout

taktran committed:
AG-13351 - Use breakpoints for hero grid

markdurrant committed:
AG-13351 - Reduce automated examples controls text size

taktran committed:
AG-13351 - Sync framework selection on homepage

* Extract framework selection hook from homepage
* Use framework selector hook for npm install script
* Update theme builder code snippet + framework selector

kylerphillips committed:
AG-13351 Improve hero spacing and copy

kylerphillips committed:
AG-13351 Fix hover in darkmode on hero button

markdurrant committed:
AG-13351 - Move `<CTAWithFrameworks />` into it's own component

kylerphillips committed:
AG-13351 Update stats to be centered on mobile

kylerphillips committed:
AG-13351 Theme builder responsive fixes

markdurrant committed:
AG-13351 - Tidy up `<CTAWithFrameworks />` component

kylerphillips committed:
AG-13351 Fix nested sass error

markdurrant committed:
AG-13351 - Reafactor `<CTAWithFrameworks />` so only hover on FW will show list

kylerphillips committed:
AG-13351 Update custom theme in personalise section

kylerphillips committed:
AG-13351 Whats new tweaks

kylerphillips committed:
AG-13351 Update showcase styling

kylerphillips committed:
AG-13351 Add em dash

markdurrant committed:
AG-13351 - `<CTAWithFrameworks />` design tweaks

taktran committed:
AG-13351 - Remove framework selector for theme builder snippet

markdurrant committed:
AG-13351 - Make homepage FAQ links respond to selected FW

taktran committed:
AG-13351 - Fix hydration issues on homepage

markdurrant committed:
AG-13351 - Homepage personalise code block, change hieght on theme

taktran committed:
AG-13351 - Fix react icon

JamesSwinton committed:
AG-13351 - Merge in copy changes

kylerphillips committed:
AG-13351 Remove extra demo link

taktran committed:
AG-13351 - Update sparkline tooltip

taktran committed:
AG-13351 - Use shared framework icons

taktran committed:
AG-13351 - Clean up code

kylerphillips committed:
AG-13351 Make whats new version cards consistent

kylerphillips committed:
AG-13351 Improve install copy animation

taktran commented on documentation/ag-grid-docs/src/components/theme-builder-homepage/ThemeBuilderHomepage.tsx:
```diff
@@ -17,25 +17,49 @@ interface Props {
-            foregroundColor: '#340c52',
-            borderColor: '#f59342',
-            chromeBackgroundColor: '#e3f5c4',
+            accentColor: '#0086F4',
```
These styles are the same now, so I'll remove the darkmode config

markdurrant committed:
AG-13351 - Fix Safari overscroll bug

taktran committed:
AG-13351 - Remove dark mode config for homepage theme builder component

taktran reviewed (approved):
LGTM 👍🏼

kylerphillips commented:
Also LGTM!

JamesSwinton committed:
Copy updates

JamesSwinton committed:
Merge branch 'AG-13351-homepage-framework-option-2' of https://github.com/ag-grid/ag-grid into AG-13351-homepage-framework-option-2

JamesSwinton committed:
Final copy update

markdurrant committed:
AG-13351 - Fix linting errors
